static/js/compose.js

```diff
@@ -628,38 +628,35 @@ export function initialize() {
         }
     });
 
-    let instance = {};
     $("body").on("click", ".time_pick", (e) => {
         e.preventDefault();
         e.stopPropagation();
 
+        $(e.target).toggleClass("has_popover");
+
         let target_textarea;
         let edit_message_id;
         if ($(e.target).parents(".message_edit_form").length === 1) {
             edit_message_id = rows.id($(e.target).parents(".message_row"));
             target_textarea = $(`#edit_form_${CSS.escape(edit_message_id)} .message_edit_content`);
         }
 
-        if (!instance.calendarContainer) {
+        if ($(e.target).hasClass("has_popover")) {
             const on_timestamp_selection = (val) => {
                 const timestr = `<time:${val}> `;
                 compose_ui.insert_syntax_and_focus(timestr, target_textarea);
             };
 
-            instance = composebox_typeahead.show_flatpickr(
-                $(e.target)[0],
+            e.target._instance = composebox_typeahead.show_flatpickr(
+                e.target,
                 on_timestamp_selection,
                 new Date(),
                 {
                     // place the time picker above the icon and centerize it horizontally
                     position: "above center",
```

Comment:
Why do we need this change?

Reply:
These changes inadvertently clashed with the previous changes. But still, I think using a css attribute is better to indicate the visibility of the popover (the same technique is used for the emoji picker).